src/ConfigResolver/index.ts

```diff
@@ -0,0 +1,91 @@
+import inclusion from 'inclusion'
+import { pathToFileURL } from 'url'
+import { ReporterContract, Refiner, NamedReporterContract } from '@japa/core'
+import { Config, NormalizedConfig } from '../Contracts'
+
+/**
+ * Wrap the value inside an array if it's not one
+ */
+function arrayify<T>(value: T | T[]): T[] {
+  return Array.isArray(value) ? value : [value]
+}
+
+/**
+ * Convert a ReporterContract object to a NamedReporterContract object
+ */
+function buildReporterConfig(reporters: ReporterContract[]): NamedReporterContract[] {
+  return reporters.map((reporter) => {
+    /**
+     * If the reporter wasn't named, then we generate a random name
+     */
+    const reporterName = reporter.name || Math.random().toString(36).substring(5)
+
+    if (typeof reporter === 'function') {
+      return { handler: reporter, name: reporterName }
+    }
+
+    return reporter
```

Comment:
Is there any reason we are assigning a random name? Maybe, we can use the `function` name as the reporter name. 

```ts
if (typeof reporter === 'function') {
  return { handler: reporter, name: report.name }
}

return reporter
```

Reply:
Yeah, basically it was to keep a unique name for each reporter

Later, they are retrieved using a `.find` method: https://github.com/japa/runner/pull/24/files#diff-dcdc3e0b3362edb8fec2a51d3fa51f8fb8af8f70247e06d9887fa934834c9122R234

Maybe we could replace that `.find` with a `.filter` but it looks weird to me. This is not something I have done yet, but eventually, I think it would be nice to display a warning if the user registers two reporters that have the same name. To avoid any error.

Reply:
I think most of the reporters will have unique names anyways. But yes, we can add validation and report duplicate reporters at the time of processing the config.

And if we validate the reporters for uniqueness, then we can use the `reporter.name` to compute the reporter name. Because, the random name is anyways not useful in any case

Reply:
Sorry, to be more clear : 

Random reporter name generation is only used when the old API is used. If a user comes to use the next @japa/runner with the old @japa/spec-reporter ( that doesn't return a named reporter, only the handler ): then we gonna attribute a random on that reporter, so that can we can enable it by default by referencing that same name later.

If by `reporter.name` you mean `Function.name` : we can't because this name will be empty if it's an anonymous function

Reply:
Yo, any update on this topic? Maybe we can talk about it on discord when we find some time, if it's not clear, it could be easier to understand each other.